### PR TITLE
[camera] Federate mobile implementations

### PR DESCRIPTION
**Pull request by stuartmorgan**
Moves the iOS and Android implementations to their own federated packages.

Temporarily marks `camera` as unpublishable to allow the implementations to be moved, rather than copied and deleted, in order to better preserve git history. A follow-up PR will restore it to publishable form.

The new `camera_controller.dart` and `camera_preview.dart` files started as copies from the same files in the app-facing package. Preview is mostly unchanged, but CameraController has been stripped down to just state tracking (no argument or state validation, no wrappers for things that are just one-line calls to the platform interface. The duplication isn't ideal, but it's just another instance of the general problem tracked by https://github.com/flutter/flutter/issues/46716, so for now we're duplicating rather than blocking federation. If we decide to extract it to a shared location later, it should be straightforward to do so.

Part of https://github.com/flutter/flutter/issues/68498

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/camera/camera_android/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## 0.9.7+1
+
+* Splits from `camera` as a federated implementation.
```
Comment (hellohuanlin) [inline packages/camera/camera_android/CHANGELOG.md:3]: do we want to add a link to the previous logs?

**Review by hellohuanlin — APPROVED** [review]
Thanks for cleaning this up. Yeah it'd be great if examples can be shared across platforms to avoid the duplicate.

File: packages/camera/camera_android/CHANGELOG.md (hunk 1 above)
Comment (stuartmorgan) [reply packages/camera/camera_android/CHANGELOG.md:3]: People only use these packages via the app-facing package, so they'll already know where that changelog is. I wouldn't expect most people to ever see anything but the app-facing package changelog.

**Comment by stuartmorgan** [comment]
I'm going to go ahead and land this despite the in-progress camera PRs around error codes. Rebasing those should be simple since the native files are moved (not copied), and this should actually make things slightly easier since each platform's error codes can be independently updated in a breaking change for just that package (as was done for web), and then the only coordination will be updating all of the implementation package versions in `camera`'s pubspec.yaml all at once.

**Pull request merged**